src/Microsoft.AspNetCore.Mvc.Core/Internal/FileContentResultExecutor.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Http.Features;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.AspNetCore.Mvc.Internal
+{
+    public class FileContentResultExecutor : FileResultExecutor
+    {
+        private readonly ILogger _logger;
+
+        public FileContentResultExecutor(ILoggerFactory loggerFactory)
+        {
+            if (loggerFactory == null)
+            {
+                throw new ArgumentNullException(nameof(loggerFactory));
+            }
+
+            _logger = loggerFactory.CreateLogger<FileContentResultExecutor>();
```

Comment:
Why isn't the `ILogger` a property in the base class? That would maintain consistent logging w/ the previous code.